geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloFeatureCollection.scala

```diff
@@ -28,16 +22,7 @@ class AccumuloFeatureCollection(source: GeoMesaFeatureSource, query: Query)
 
   override def accepts(visitor: FeatureVisitor, progress: ProgressListener): Unit =
     visitor match {
-      case v: TubeVisitor      => v.setValue(v.tubeSelect(source, query))
-      case v: ProximityVisitor => v.setValue(v.proximitySearch(source, query))
-      case v: QueryVisitor     => v.setValue(v.query(source, query))
-      case v: StatsVisitor     => v.setValue(v.query(source, query))
-      case v: SamplingVisitor  => v.setValue(v.sample(source, query))
```

Comment:
yeah, everything goes through query hints - then it's up to the data store impls to respect the hints